client/components/SpikeEventForm/index.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Field, reduxForm } from 'redux-form'
+import { RelatedPlannings } from '../index'
+import * as actions from '../../actions'
+import moment from 'moment'
+import { EventUpdateMethodField, EventUpdateMethods } from '../fields/index'
+import './style.scss'
+
+class Component extends React.Component {
+
+    constructor(props) {
+        super(props)
+        this.state = { plannings: props.initialValues._plannings }
+    }
+
+    onUpdateMethodChange(method) {
+        let event = this.props.initialValues
+        let events = event._recurring.events
+        let storedPlannings = event._recurring.plannings
+        let plannings
+
+        // Spike selected event only
+        if (method.value === EventUpdateMethods[0].value) {
+            plannings = Object.keys(storedPlannings).filter((pid) => (
+                storedPlannings[pid].event_item === event._id
+            )).map((pid) => ({ ...storedPlannings[pid] }))
+
+        // Spike selected and future events
+        } else if (method.value === EventUpdateMethods[1].value) {
+            let ids = events.filter((e) => (
+                moment(e.dates.start).isSameOrAfter(moment(event.dates.start))
+            )).map((e) => (e._id)) || []
+            plannings = Object.keys(storedPlannings).filter((pid) => (
+                (ids || []).indexOf(storedPlannings[pid].event_item) > -1
+            )).map((pid) => ({ ...storedPlannings[pid] }))
+
+        // Spike all events
+        } else if (method.value === EventUpdateMethods[2].value) {
+            plannings = Object.keys(storedPlannings).filter((pid) => (
+                (event._recurring.ids || []).indexOf(storedPlannings[pid].event_item) > -1
+            )).map((pid) => ({ ...storedPlannings[pid] }))
+
+        } else {
+            throw Error('Incorrect update method: ' + method)
+        }
+
+        this.setState({ plannings })
+        this.props.initialValues._plannings = plannings
```

Comment:
I will convert this component to a `stateless functional component`, and move the filtering logic into a `reducer`.